### ISSUE TITLE
Add TemplatingException for errors that occur with plain SQL templating

### ISSUE THREAD

**Issue by MikePryadko** [jOOQ/jOOQ#10493]
### Actual behaviour
```
Caused by: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at org.jooq_3.13.1.POSTGRES.debug(Unknown Source) ~[na:na]
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[na:na]
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[na:na]
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248) ~[na:na]
	at java.base/java.util.Objects.checkIndex(Objects.java:372) ~[na:na]
	at java.base/java.util.ArrayList.get(ArrayList.java:459) ~[na:na]
	at org.jooq.impl.Tools.renderAndBind(Tools.java:2504) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.SQLImpl.accept(SQLImpl.java:66) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:591) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.AbstractContext.visit0(AbstractContext.java:457) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:218) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.SQLField.accept(SQLField.java:65) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:591) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.AbstractContext.visit0(AbstractContext.java:457) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:218) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.FieldMapForUpdate.accept(FieldMapForUpdate.java:113) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:591) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.AbstractContext.visit0(AbstractContext.java:457) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:218) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.InsertQueryImpl.accept0(InsertQueryImpl.java:400) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.AbstractDMLQuery.accept(AbstractDMLQuery.java:565) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:591) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.AbstractContext.visit0(AbstractContext.java:457) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:218) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.AbstractQuery.getSQL0(AbstractQuery.java:523) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:326) ~[jooq-3.13.1.jar:na]
	at org.jooq.impl.InsertImpl.fetchOne(InsertImpl.java:1059) ~[jooq-3.13.1.jar:na]
        ...
```
### Expected behaviour
Something like `QueryPartsCountMismatchException: expected 2, found: 0, field: "{0} test {1}"` or some checks on project compilation stage or some plugin for Intellij Idea for highlighting (like it works with slf4j)
![tmp](https://user-images.githubusercontent.com/31248040/90331230-874bed80-dfbb-11ea-91d9-426d76268e68.png)

### Steps to reproduce the problem
```java
DSL.select(DSL.field("{0} test {1}", String.class)).getSQL();
```
### Versions
- jOOQ: 3.13.1
- Java: openJDK 11
- Database (include vendor): PostgreSQL 12
- OS: Linux

**Comment by lukaseder** [comment]
Thanks a lot for your suggestion. I agree, there could be a more specific exception than the one we currently have.

**Comment by lukaseder** [comment]
> some checks on project compilation stage or some plugin for Intellij Idea for highlighting (like it works with slf4j)

I really like that idea. Will add it to this feature request: https://github.com/jOOQ/jOOQ/issues/8359

**Comment by lukaseder** [comment]
Fixed for 3.14.0.
